tests/test_sqlite_map.py

```diff
@@ -19,40 +21,105 @@ def setUp(self):
         setup_testdb(self.db)
         self.reader = ExampleMapReader(self.db)
 
+    def test_line_invalid_id_type(self):
+        "Check if an invalid line id raises an error"
+        with self.assertRaises(ExampleMapError):
+            Line(self.reader, "Hello")
+
+    def test_line_nonexistent_id(self):
+        "Check if a nonexistent line ID leads to an exception"
+        with self.assertRaises(TypeError):
+            _ = Line(self.reader, 20).length
+        with self.assertRaises(ExampleMapError):
+            self.reader.get_line(20)
+
     def test_line_length(self):
         "Check the length of the line with ID 1"
         line = self.reader.get_line(1)
-        self.assertAlmostEqual(line.length, 391, delta=1)
+        self.assertAlmostEqual(line.length, 271, delta=1)
 
     def test_linepoints(self):
         "Test the point count of every line"
         for line in self.reader.get_lines():
-            self.assertEqual(line.num_points(), 2)
+            if line.line_id == 18:
```

Comment:
Maybe this can be cleaned up.  You could have a mapping between the line IDs and expectations and loop over and check.